epoxy-paging/src/main/java/com/airbnb/epoxy/paging/CachingPagingEpoxyController.kt

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.airbnb.epoxy.paging
+
+import android.arch.paging.PagedList
+import android.os.Handler
+import android.support.v7.util.DiffUtil
+import com.airbnb.epoxy.EpoxyController
+import com.airbnb.epoxy.EpoxyModel
+import com.airbnb.epoxy.EpoxyViewHolder
+
+/**
+ * An [EpoxyController] that can work with a [PagedList].
+ *
+ * Internally, it caches the model for each item in the [PagedList]. You should override
+ * [buildItemModel] method to build the model for the given item. Since [PagedList] might include
+ * `null` items if placeholders are enabled, this method needs to handle `null` values in the list.
+ *
+ * By default, the model for each item is added  to the model list. To change this behavior (to
+ * filter items or inject extra items), you can override [addModels] function and manually add built
+ * models.
+ */
+abstract class CachingPagingEpoxyController<T>(
```

Comment:
I'm not happy with this name actually :/ the caching part bothers me.
Should we call it `PagedListEpoxyController`? I didn't go with it since there is `PagingEpoxyController`

Reply:
@yigit sure, I like `PagedListEpoxyController`. I am in favor of getting rid of `PagingEpoxyController` eventually once this is adopted